server/service/src/rnr_form/generate_rnr_form_lines.rs

```diff
@@ -1,95 +1,339 @@
+use std::{collections::HashMap, ops::Neg};
+
+use chrono::{Duration, NaiveDate};
 use repository::{
-    EqualFilter, MasterListLineFilter, MasterListLineRepository, PeriodRow, RepositoryError,
-    RnRFormLineRow,
+    AdjustmentFilter, AdjustmentRepository, ConsumptionFilter, ConsumptionRepository, DateFilter,
+    DatetimeFilter, EqualFilter, MasterListLineFilter, MasterListLineRepository, Pagination,
+    PeriodRow, ReplenishmentFilter, ReplenishmentRepository, RepositoryError, RnRForm,
+    RnRFormLineRow, RnRFormLineRowRepository, StockLineFilter, StockLineRepository, StockLineSort,
+    StockLineSortField, StockMovementFilter, StockMovementRepository, StockOnHandFilter,
+    StockOnHandRepository, StorageConnection,
 };
-use util::uuid::uuid;
+use util::{constants::NUMBER_OF_DAYS_IN_A_MONTH, date_now, date_with_offset, uuid::uuid};
 
 use crate::{
-    item_stats::{get_item_stats, ItemStatsFilter},
+    requisition_line::chart::{get_stock_evolution_for_item, StockEvolutionOptions},
     service_provider::ServiceContext,
 };
 
+// Would be nice if this was an OMS store pref
+const TARGET_MOS: f64 = 2.0;
+
 pub fn generate_rnr_form_lines(
     ctx: &ServiceContext,
+    store_id: &str,
     rnr_form_id: &str,
     master_list_id: &str,
     period: PeriodRow,
+    previous_form: Option<RnRForm>,
 ) -> Result<Vec<RnRFormLineRow>, RepositoryError> {
-    let master_list_lines = MasterListLineRepository::new(&ctx.connection).query_by_filter(
-        MasterListLineFilter::new().master_list_id(EqualFilter::equal_to(master_list_id)),
+    let master_list_item_ids = get_master_list_item_ids(&ctx, master_list_id)?;
+
+    let period_length_in_days = get_period_length(&period);
+    let lookback_months = period_length_in_days as f64 / NUMBER_OF_DAYS_IN_A_MONTH;
```

Comment:
This is what we have done for AMC throughout the rest of the app... however, there was mention in the R&R forms of getting actual days per month for the period - should I do that here?